### PR TITLE
Add link checking for step-by-step pages

### PR DESCRIPTION
**Pull request by leenagupte**
Trello: https://trello.com/c/F1S78A0L 

## What's changed

Give publishers the ability to check for broken links in a step by step page. The link error messages appear against each step so they know where to go to fix the broken link.

This change uses [link-checker-api](https://docs.publishing.service.gov.uk/apis/link-checker-api.html#content) to check links in a similar way to the Mainstream Publisher app.

### Overview page
screen shot 2018-08-30 at 08 44 36

screen shot 2018-08-30 at 08 45 09

#### Collapsed
44145144-53337b8c-a082-11e8-855a-cdc60c94340b

#### Open
44145148-57cb3c34-a082-11e8-8c78-f31b3afaed23

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -12,7 +12,25 @@
 
 ActiveRecord::Schema.define(version: 2018_08_17_100339) do
 
-  create_table "list_items", id: :integer, force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8" do |t|
+  create_table "link_check_reports", options: "ENGINE=InnoDB DEFAULT CHARSET=latin1", force: :cascade do |t|
```
Comment (leenagupte) [inline db/schema.rb]: @jonodrew Do we ever use this table?

Comment (jonodrew) [reply db/schema.rb]: No, I thought I'd removed it in an earlier commit. Not sure how that's sneaked back in there.

Comment (leenagupte) [reply db/schema.rb]: I'll get rid of it and fix-up.

File: app/assets/stylesheets/steps.scss [hunk 2]
```diff
@@ -76,3 +77,39 @@ $red: #b10e1e;
...
+  text-decoration: none;
+  color: $beige;
+
+  &:hover, &:focus {
+    text-decoration: none;
+    color: $beige;
+  }
+}
+
+.broken-links-accordion__indicator, {
```
Comment (vanitabarrett) [inline app/assets/stylesheets/steps.scss]: Stray `,`?

File: app/views/shared/steps/_broken-links-summary.html.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<span class="glyphicon glyphicon-warning-sign" aria-hidden="true"></span>
+<strong><%= step.broken_links.count %></strong> broken link<%= "(s)" if step.broken_links.count > 1 %> found
```
Comment (vanitabarrett) [inline app/views/shared/steps/_broken-links-summary.html.erb]: What is the reasoning behind using `<strong>` here, to only wrap the number?

File: app/views/step_by_step_pages/show.html.erb [hunk 4]
```diff
@@ -73,5 +73,8 @@
         <li><%= link_to 'Preview', preview_url(@step_by_step_page.slug, auth_bypass_id: @step_by_step_page.auth_bypass_id), class: "btn btn-success btn-block preview-button step-action" %></li>
       </ul>
     </div>
+    <div>
+        <%= button_to "Check for broken links", {action: "check_links", step_by_step_page_id: @step_by_step_page.id}, class: "btn btn-primary" %>
```
Comment (vanitabarrett) [inline app/views/step_by_step_pages/show.html.erb]: Think this is indented too far?

File: app/assets/stylesheets/steps.scss [hunk 5]
```diff
@@ -6,6 +6,7 @@ $light-grey: #f5f5f5;
 $light-grey-2: #f8f8f8;
 $black: #0B0C0C;
 $red: #b10e1e;
+$grey: #eee;
```
Comment (vanitabarrett) [inline app/assets/stylesheets/steps.scss]: This isn't really important, but for consistency with other colours we might want to use the full code? And decide on caps/no caps.

Comment (leenagupte) [reply app/assets/stylesheets/steps.scss]: @vanitabarrett Is the full code #eeeeee?

Comment (vanitabarrett) [reply app/assets/stylesheets/steps.scss]: @leenagupte Yep!

File: app/views/shared/steps/_broken-links-summary.html.erb (hunk 3 above)
Comment (leenagupte) [reply app/views/shared/steps/_broken-links-summary.html.erb]: cc @DilwoarH

Comment (DilwoarH) [reply app/views/shared/steps/_broken-links-summary.html.erb]: Not sure why this was done, maybe we should remove it or wrap it around the whole text

Comment (leenagupte) [reply app/views/shared/steps/_broken-links-summary.html.erb]: @DilwoarH I'll wrap the whole thing and see what it looks like.

**Pull request merged**